Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -17,38 +17,85 @@
+    and :math:`\mathrm{prox}_{C}` is the projection over a set :math:`C`.
+
+    Parameters
+    ----------
+
+    initial : DataContainer, default = None
+              Starting point of the algorithm, default value = Zero DataContainer 
+    operator : LinearOperator
+              The operator A.
+    data : DataContainer
+           The data b.
+    lower : :obj:`float`, default = None
+            Lower bound constraint, default value = :code:`-inf`.
+    upper : :obj:`float`, default = None
+            Upper bound constraint, default value = :code:`-inf`.
+    constraint : Function, default = None
+                A function with :code:`proximal` method, e.g., :class:`.IndicatorBox` function and :meth:`.IndicatorBox.proximal`,
+                or :class:`.TotalVariation` function and :meth:`.TotalVariation.proximal`.
+
+    kwargs:
+        Keyword arguments used from the base class :class:`.Algorithm`.    
+
+    Note 
+    ----
+    
+    If :code:`constraint` is not passed, then :code:`lower` and :code:`upper` are looked at and an :class:`.IndicatorBox`
+    function is created.
+
+    If :code:`constraint` is passed, :code:`proximal` method is required to be implemented.
+
+    If :math:`M = \frac{1}{A*\mathbb{1}}`, :math:`D = \frac{1}{A^{T}\mathbb{1}}` contain :code:`NaN`, :math:`\pm\inf`
+    they are replaced by :math:`1`.
+
 
-    r'''Simultaneous Iterative Reconstruction Technique
     
-    Problem: 
+
     
-    .. math::  
     
-      A x = b
+    Examples
+    --------
+    .. math:: \underset{x}{\mathrm{argmin}} \| x - 5\|^{2}
     
-    :param initial: Initial guess
-    :param operator: Linear operator for the inverse problem
-    :param data: Acquired data to reconstruct       
-    :param constraint: Function proximal method
-                e.g.  :math:`x\in[0, 1]`, :code:`IndicatorBox` to enforce box constraints
-                        Default is :code:`None`).
-    '''
-    def __init__(self, initial=None, operator=None, data=None, lower=None, upper=None, constraint=None, **kwargs):
-        '''SIRT algorithm creator
-
-       Optional parameters:
-
-      :param initial: Initial guess
-      :param operator: Linear operator for the inverse problem
-      :param data: Acquired data to reconstruct 
-      :param lower: Scalar specifying lower bound constraint on pixel values, default -inf
-      :param upper: Scalar specifying upper bound constraint on pixel values, default +inf
-      :param constraint: More general constraint, given as Function proximal method
-                   e.g.  :math:`x\in[0, 1]`, :code:`IndicatorBox` to enforce box constraints
-                         Default is :code:`None`). constraint takes priority over lower and upper.'''
+    >>> from cil.framework import ImageGeometry
+    >>> from cil.optimisation.operators import IdentityOperator    
+    >>> from cil.optimisation.algorithms import SIRT
+    >>> ig = ImageGeometry(3,4)
```

Comment:
I believe these examples with runnable code we need to discuss. I see the argument for them. but it's likely that in the future we'll change things and never look here to update the snippets.

`ImageGeometry` is a prime candidate as the signature `(voxel_num_x, voxel_num_y,voxel_num_z)` is clunky and doesn't make a lot of sense. When we make those changes this will be using deprecated code.

Maybe as a compromise it's better to use something from `dataexamples` with a `get` method that is less likely to change.

Reply:
`ImageGeometry` is the 3rd most important class in cil. 1st is `DataContainer`, 2nd is `ImageData`. If we break `ImageGeometry` then we will definitely know it. This example solve the min of \|x - 5\|^{2}, x = 5 so it is a very trivial example. Atm, we do not have a lot of `Examples` but we should. And if we want to check our `Examples` script then we have to do [doctest](https://docs.python.org/3/library/doctest.html#unittest-api)